ansible_rulebook/job_template_runner.py

```diff
@@ -47,41 +46,33 @@ def __init__(
         self.refresh_delay = int(
             os.environ.get("EDA_JOB_TEMPLATE_REFRESH_DELAY", 10)
         )
-
-    async def _get_page(
-        self, session: aiohttp.ClientSession, href_slug: str, params: dict
-    ) -> dict:
-        url = urljoin(self.host, href_slug)
-        async with session.get(
-            url, params=params, ssl=self._sslcontext
-        ) as response:
-            response_text = dict(
-                status=response.status, body=await response.text()
-            )
-        if response_text["status"] != 200:
-            raise ControllerApiException(
-                "Failed to get from %s. Status: %s, Body: %s"
-                % (
-                    url,
-                    response_text["status"],
-                    response_text.get("body", "empty"),
-                )
+        self._session = None
+
+    def _create_session(self):
```

Comment:
I think the session should be closed at the end. 
I suggest:
```
      def _close_session(self):
          if self._session and not self._closed:
              self._session.close()
```
```
            self._session = aiohttp.ClientSession(
                connector=aiohttp.TCPConnector(limit=limit),
                headers=self._auth_headers(),
                raise_for_status=True,
            )
            atexit.register(self._close_session)
```

It requires [atexit](https://docs.python.org/3/library/atexit.html) module

Reply:
My suggested approach is not the most optimal, because the object will keep in memory until the end of the program and the garbage collector will not be able to collect it. But I think it is minor and I can not imagine an easy way to do it better.